test/unit-tests/openBrowser.test.js

```diff
@@ -23,6 +32,173 @@ describe('opens browser successfully', () => {
   afterEach(async () => await closeBrowser());
 });
 
+describe('open browser and create browser context', () => {
+  let url1;
+  let url2;
+  it('Should have incognito window', async () => {
+    let innerHtml = `<section class="header">
+      <h1>Incognito tests</h1>
+        </section>
+          <section class='main-content'>
+            <div class='item'>
+               Item 1
+            </div>
+            <div class='item'>
+               Browser1
+         </div>
+      </section>
+`;
+
+    let innerHtml1 = `<section class="header">
+<h1>Incognitotests</h1>
+  </section>
+    <section class='main-content'>
+      <div class='item'>
+         Item 2
+      </div>
+      <div class='item'>
+      Browser2
+</div>
+</section>
+`;
+    url1 = createHtml(innerHtml, 'Incognito');
+    await openBrowser();
+    setConfig({
+      waitForNavigation: true,
+      retryTimeout: 100,
+      retryInterval: 10,
+    });
+    await openWindow(url1, { name: 'admin', incognito: true });
```

Comment:
### Test case 5

```
Version: 1.0.3 (Chromium:81.0.3994.0)
Type .api for help and .exit to quit

> openBrowser()
 ✔ Browser opened
> openWindow("gauge.org", { name : "gauge"})
 ✔ Window opened with name gauge
> openWindow("gauge.org", { name : "gauge"})
 ✔ Window opened with name gauge
> openWindow("gauge.org", { name : "gauge"})
 ✔ Window opened with name gauge
> switchTo("gauge")
 ✘ Error: No tab(s) matching gauge found, run `.trace` for more info.
> closeWindow("gauge")
 ✔ Window with name gauge closed
```

#### Observation

Should warn the user and handle the case in case the same name is used for windows. Again in this case warn the user and assign a random name and print the assigned name.
Also noticed that `closeWindow` does not work.

Reply:
### Test case 6

```
Version: 1.0.3 (Chromium:81.0.3994.0)
Type .api for help and .exit to quit

> openBrowser()
 ✔ Browser opened
> openWindow("gauge.org", { name: "gauge"})
 ✔ Window opened with name gauge
> openWindow("taiko.gauge.org", { name: "taiko"})
 ✔ Window opened with name taiko
> switchTo({name: "gauge"})
 ✔ Switched to Incognito window gauge
>  ✔ Switched to Incognito window matching [object Object]
>
```

#### Observation

The message is wrong should've been
For normal window

```
> switchTo({name: "gauge"})
 ✔ Switched to window "gauge"
```

For incognito window

```
> switchTo({name: "gauge"})
 ✔ Switched to incognito window "gauge"
```

Reply:
> The REPL opens and window and hangs after `openWIndow()` is run without parameters after `openBrowser`

Will throw an Error in this case?

Reply:
> > openBrowser()
>  ✔ Opened browser
> > openWindow("gauge.org")
>  ✔ Opened window with name window1

@zabil This is a new change in existing `openBrowser ` API.

Reply:
> This is a new change in existing openBrowser API.

Sorry for the late response. Yeah that is fine. I assume it only involved changing that message.